### PR TITLE
When you run a container with ceph it will fail with the error:

### PR DESCRIPTION
**Pull request by a-nldisr**
```
failed: '/bin/systemd-run -r bash -c 'ulimit -n 32768; /usr/bin/ceph-osd -i 2 --pid-file /var/run/ceph/osd.2.pid -c /etc/ceph/ceph.conf --cluster ceph -f''
Failed to create bus connection: No such file or directory
```

It's unlikely that people are running systemd within containers and thus i added a check if the processes is running in a container when systemd-run is triggered, if docker is detected the usual startup process is executed.

I chose to use grep -q to detect docker because it works on every system (tested it on privileged containers).

Signed-off-by: Rogier Dikkes <r.dikkes@gmail.com>

### CODE REVIEW

File: src/init-ceph.in [hunk 1]
```diff
@@ -21,6 +21,11 @@ fi
 SYSTEMD_RUN=$(which systemd-run 2>/dev/null)
 grep -qs systemd /proc/1/comm || SYSTEMD_RUN=""
 
+# Detect if this is inside docker, systemd-run fails inside containers.
+function detectDocker {
+readonly DOCKER_RUN=$(grep -q docker /proc/1/cgroup)
```
Comment (leseb) [inline src/init-ceph.in]: :-1: This doesn't work on my system. I believe checking the existence of `/.dockerinit` will always work.

File: src/init-radosgw [hunk 2]
```diff
@@ -23,6 +23,11 @@ else
     grep -qs systemd /proc/1/comm || SYSTEMD_RUN=""
 fi
 
+# Detect Docker, if we run systemd-run within docker the process will fail.
+function detectDocker {
+readonly DOCKER_RUN=$(grep -q docker /proc/1/cgroup)
```
Comment (leseb) [inline src/init-radosgw]: :-1: This doesn't work on my system. I believe checking the existence of `/.dockerinit` will always work.

**Comment by a-nldisr** [comment]
Sorry for this, for some reason im messing up with amend. Ill fix it.

**Comment by a-nldisr** [comment]
Dear god, how stupid can i get... Sorry for this im closing this pull, sort things out and create a new pull request without all these stupid mistakes.

**Comment by leseb** [comment]
@a-nldisr sure no problem at all :), looking forward to seing your next PR

**Comment by a-nldisr** [comment]
https://github.com/ceph/ceph/pull/7600 opened

**Pull request closed without merging**